cache/ttl_cache.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+import type { MemoizationCache } from "./memoize.ts";
+import { delay } from "@std/async/delay";
+
+/**
+ * Time-to-live cache.
+ *
+ * Automatically removes entries once the configured amount of time elapses. If
+ * the values themselves are promises, the countdown starts from when they
+ * resolve.
```

Comment:
Hm... I'm not sure that promises should get any special treatment... @oles, do you have any preferences?

Reply:
That sounds a good point. I don't see the rationale for the special handling of `Promise` very well.

Also this seems causing uncaught error when the promise is rejected. That is probably inconvenient for most users.